modules/core/src/main/java/org/apache/ignite/metric/IgniteMetrics.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.metric;
+
+import org.apache.ignite.lang.IgniteExperimental;
+import org.apache.ignite.spi.metric.ReadOnlyMetricRegistry;
+
+/**
+ * Allows to manage custom metrics.
+ * <p>
+ * Metrics are grouped into registries (groups). Every metric has full name which is the conjunction of registry name
+ * and the metric short name. Within a registry metric has only its own short name.
+ * <p>
+ * Note: Names of custom metric registries are required to start with 'custom.' (lower case) and may have additional
```

Comment:
Can we go with only one logic - either add `custom.` to all provided names or check if registryName starts with `custom.` and fail if not? I prefer first option - let's just add `custom.` to all provided `registryName`.